### DISCUSSION TITLE
Where are onSaveGame and onLoadGame implemented?

### DISCUSSION THREAD

**Discussion by Newbytee** [Try/OpenGothic#331] [Q&A] [answered]
Hi! I'm working on some filesystem-related changes to OpenGothic, and I'm trying to figure out how saving and loading saves works. I found that onLoadGame and onLoadSave are called to do this, and I found their prototypes here: https://github.com/Try/OpenGothic/blob/d5b45f9e2eba510e817085697e64ba07764cfad4/game/gothic.cpp#L486-L492

However, I don't see where these are implemented. Where do I find their implementations?

**Comment by Try** [comment, marked as answer]
Hi, @Newbytee !

Those are signals, connected here:
 https://github.com/Try/OpenGothic/blob/d5b45f9e2eba510e817085697e64ba07764cfad4/game/mainwindow.cpp#L60-L66

As for  file-system related stuff:
https://github.com/Try/OpenGothic/blob/d5b45f9e2eba510e817085697e64ba07764cfad4/game/mainwindow.cpp#L878-L884

`Tempest::RFile` is for reading, `WFile` for writing
